samples/Suppressions.sarif

```diff
@@ -0,0 +1,130 @@
+{
+  "$schema": "https://schemastore.azurewebsites.net/schemas/json/sarif-2.1.0-rtm.5.json",
+  "version": "2.1.0",
+  "properties": {
+    "comment": "This sample demonstrates how suppressions affect the visibility of results in SARIF viewers."
+  },
+  "runs": [
+    {
+      "tool": {
+        "driver": {
+          "name": "SarifSamples"
+        }
+      },
+      "results": [
+        {
+          "ruleId": "SMP0001",
+          "message": {
+            "text": "This result is visible because it is not suppressed."
+          }
+        },
+        {
+          "ruleId": "SMP0001",
+          "message": {
+            "text": "This result is hidden because it is suppressed in source."
+          },
+          "suppressions": [
+            {
+              "kind": "inSource"
+            }
+          ]
+        },
+        {
+          "ruleId": "SMP0001",
+          "message": {
+            "text": "This result is hidden because it is suppressed externally."
+          },
+          "suppressions": [
+            {
+              "kind": "external"
+            }
+          ]
+        },
+        {
+          "ruleId": "SMP0001",
+          "message": {
+            "text": "This result is visible because its suppression was rejected."
+          },
+          "suppressions": [
+            {
+              "kind": "external",
+              "status": "rejected"
+            }
+          ]
+        },
+        {
+          "ruleId": "SMP0001",
+          "message": {
+            "text": "This result is hidden because its suppression was accepted."
+          },
+          "suppressions": [
+            {
+              "kind": "external",
+              "status": "accepted"
+            }
+          ]
+        },
+        {
+          "ruleId": "SMP0001",
+          "message": {
+            "text": "This result is visible because its suppression is still under review."
+          },
+          "suppressions": [
+            {
+              "kind": "external",
+              "status": "underReview"
+            }
+          ]
+        },
+        {
+          "ruleId": "SMP0001",
+          "message": {
+            "text": "This result is hidden because at least one suppression was accepted."

```

Comment:
>accepted [](start = 80, length = 8)

From here on down, reasonable people could disagree on the treatment. Interested in your opinions.

Reply:
You also need to consider what it means to have a suppression with no status in the set. Do we regard that as implicitly accepted? I think we should.

IOW same case as you have here, but remove the status of accepted for that case. I'd argue that it's the same situation, we need to regarded this as suppressed. Let's add this other case.

---
In reply to: [412508048](https://github.com/microsoft/sarif-tutorials/pull/12#discussion_r412508048) [](ancestors = 412508048)

Reply:
I do have that case covered. See the result starting on line 95 ("unqualified suppression").

---
In reply to: [413338104](https://github.com/microsoft/sarif-tutorials/pull/12#discussion_r413338104) [](ancestors = 413338104,412508048)

Reply:
and so you do. thanks.

---
In reply to: [413339480](https://github.com/microsoft/sarif-tutorials/pull/12#discussion_r413339480) [](ancestors = 413339480,413338104,412508048)